.github/scripts/build-ubuntu.sh

```diff
@@ -0,0 +1,112 @@
+[[ -z $OS_VERSION ]] && OS_VERSION="$(grep 'VERSION_ID=' /etc/os-release | cut -f2 -d'=' | tr -d '"')"
+
+# Get the base Lutris version in the same way that the Makefile does.
+[[ -z $LUTRIS_VERSION ]] && LUTRIS_VERSION="$(grep "__version__" "${CODEBASE_ROOT}/lutris/__init__.py" | cut -d" " -f 3 | sed 's|"\(.*\)"|\1|')"
+
+# Creates a GPG keyring using the key passed from the GitHub workflow.
+if [[ -z $PPA_GPG_KEY_ID ]]; then
+    echo "::group::Importing GPG private key..."
+    PPA_GPG_KEY_ID=$(echo "${PPA_GPG_PRIVATE_KEY}" | gpg --import-options show-only --import | sed -n '2s/^\s*//p')
+    export PPA_GPG_KEY_ID
+    echo "${PPA_GPG_KEY_ID}"
+    echo "${PPA_GPG_PRIVATE_KEY}" | gpg --batch --passphrase "${PPA_GPG_PASSPHRASE}" --import
+    echo "::endgroup::"
+
+    # May as well since we don't need after at this point.
+    unset PPA_GPG_PRIVATE_KEY
+fi
+
+# Add the ppa if it isn't already.
+if ! apt-add-repository -L | grep -qi "${PPA_URI//ppa:/}"; then
+    sudo apt-add-repository -y "${PPA_URI}"
+fi
+# Get the current version of Lutris on the PPA.
+if APT_VERSION="$(apt show lutris | grep 'Version: ' | cut -f2 -d' ')"; then
+    echo "Latest version on '${PPA_URI}' is '${APT_VERSION}'"
+else
+    echo "Pushing first package to '${PPA_URI}'"
+fi
+
+# Version numbers are recommended to follow the guide at:
+# https://help.launchpad.net/Packaging/PPA/BuildingASourcePackage#Versioning
+#
+# The basic format is:
+# <lutris_version>-<lutris_revision>ubuntu<ubuntu_specific_revision>ppa<ppa_revision>~ubuntu<ubuntu_version>
+#
+# ex. 0.5.12-0ubuntu1 (for just the package version)
+# ex. 0.5.12-0ubuntu1ppa1~ubuntu22.04 (for a package version meant for jammy)
+# ex. 0.5.12-0ubuntu1ppa1~ubuntu20.04 (for a package version meant for focal)
+# etc...
+#
+PPA_VERSION="ppa1~ubuntu${OS_VERSION}"
+
+# If the Lutris version doesn't have a revision, we add revision 0.
+LUTRIS_DEBIAN_VERSION="${LUTRIS_VERSION}"
+if [[ "${LUTRIS_VERSION}" = "${LUTRIS_VERSION/-*/}" ]]; then
+    LUTRIS_DEBIAN_VERSION="${LUTRIS_DEBIAN_VERSION}-0"
+fi
+
+# Finally, add an ubuntu revision, so that other packages can override ours
+# without bumping the actual version number.
+LUTRIS_DEBIAN_VERSION="${LUTRIS_DEBIAN_VERSION}ubuntu1"
+
+# If the version we're currently building exists on the PPA, increment
+# the PPA version number so that we supersede it.  Assuming a version
+# scheme like 0.5.12-0ubuntu1ppa1~ubuntu22.04, we trim everything up to
+# and including 'ppa' and everything after and including '~' to get the
+# PPA version number.
+if [[ "${APT_VERSION//ppa*/}" = "${LUTRIS_DEBIAN_VERSION}" ]]; then
+    echo "PPA (${PPA_URI}) has matching package version: ${LUTRIS_DEBIAN_VERSION}${PPA_VERSION}"
+    APT_PPA_VERSION="${APT_VERSION//*ppa/}"
+    APT_PPA_VERSION="${APT_PPA_VERSION//~*/}"
+    (( APT_PPA_VERSION++ ))
+    PPA_VERSION="ppa${APT_PPA_VERSION}~ubuntu${OS_VERSION}"
+fi
```

Comment:
Fancy!

I think this is the code to handle the mess with 0.5.12, where @strycore reverted a change and replaced the .deb just after release, to fix a changed dependency that broke Lutris for Debian.

I should comment that this sort of thing seems to be quite rare, and a more manual process might be adequate.

Reply:
Commit https://github.com/lutris/lutris/pull/4653/commits/c14d45f79006df85c076afb52928da629a9fbbd3 should prevent autoincrement logic when the PPA number is either missing or in an unexpected format, in which case `ppa1` will be used.